src/main/java/com/wavefront/WavefrontTraceLoader.java

```diff
@@ -17,51 +16,53 @@
  * @author Sirak Ghazaryan (sghazaryan@vmware.com)
  */
 public class WavefrontTraceLoader extends AbstractTraceLoader {
-  private SpanGenerator spanGenerator;
-  private SpanSender spanSender;
-  private SpanQueue spanQueue;
+  protected final SpanQueue spanQueue = new SpanQueue();
+  protected SpanGenerator spanGenerator;
+  protected SpanSender spanSender;
 
   public static void main(String[] args) throws IOException {
     new WavefrontTraceLoader().start(args);
   }
 
   @Override
   void setupSenders() throws IOException {
-    ApplicationConfig config = loadApplicationConfig();
-    if (config == null) {
+    if (applicationConfig == null) {
       throw new IOException("Application config should contain proxy or direction ingestion info.");
     }
```

Comment:
this check is not needed anymore

Reply:
It is possible that applicationConfig will not be instantiated before this point.